packages/amplify-util-uibuilder/src/utils/printDetachedFormsWarning.ts

```diff
@@ -0,0 +1,54 @@
+
+
+import { $TSContext } from 'amplify-cli-core';
+import type { Schema } from '@aws-amplify/datastore';
+import { printer } from 'amplify-prompts';
+import AmplifyUIBuilder from 'aws-sdk/clients/amplifyuibuilder';
+import { AmplifyStudioClient } from '../clients';
+import { isFormDetachedFromModel, isFormSchemaCustomized, shouldRenderComponents } from '../commands/utils';
+
+/**
+ * Handles showing a warning for detached forms pre-push.
+ */
+export const printDetachedFormsWarning = async (context: $TSContext): Promise<void> => {
+  if (!(await shouldRenderComponents(context))) {
+    return;
+  }
+  try {
+    const studioClient = await AmplifyStudioClient.setClientInfo(context);
+    const [formSchemas, localSchema] = await Promise.all<[Promise<{
+      entities: AmplifyUIBuilder.Form[];
+  }>, Promise<Schema>]>([
+    studioClient.listForms(),
+    context.amplify.invokePluginMethod(
+      context,
+      'codegen',
+      undefined,
+      'getModelIntrospection',
+      [context],
+    ),
+  ]);
```

Comment:
This is a lot of heavy lifting for a function called `printDetachedFormsWarning`. Can this be broken into separate functions?

Reply:
What if I rename it to handlePrePush?

Reply:
Is it just the name that's an issue? As of right now, the only thing this function does is print the warning. What other helper functions were you thinking? We have them for getting the forms, schema, isFormDetachedFromModel and isFormSchemaCustomized. Trying to think of another helper that wouldn't just be a thin wrapper...

Reply:
Sorry for not being clearer in my feedback. I was thinking of separating the printing logic into it's own functions that takes `formSchemas` as a parameter.

Something like:
```ts
export const prePushHandler = async (context: $TSContext): Promise<void> => {
  if (!(await shouldRenderComponents(context))) {
    return;
  }
  try {
    const studioClient = await AmplifyStudioClient.setClientInfo(context);
    const [formSchemas, localSchema] = await Promise.all<[Promise<{
      entities: AmplifyUIBuilder.Form[];
  }>, Promise<Schema>]>([
    studioClient.listForms(),
    context.amplify.invokePluginMethod(
      context,
      'codegen',
      undefined,
      'getModelIntrospection',
      [context],
    ),
  ]);

    if (!localSchema) {
      printer.debug('Local schema not found');
      return;
    }
    printDetachedFormsWarning(formSchema, localSchema)
  } catch (error) {
    printer.debug(error);
  }
}

const printDetachedFormsWarning = (formSchema: SchemaType, localSchema: SchemaType): void => {
  const modelNames = new Set(Object.keys(localSchema.models));
  const detachedCustomForms: string[] = [];

  formSchemas.entities.forEach(form => {
    // Only show warnings for schema's that have customizations, otherwise it's the same as
    // an autogen form.
    if (isFormDetachedFromModel(form, modelNames) && isFormSchemaCustomized(form)) {
      detachedCustomForms.push(form.name);
    }
  });

  if (detachedCustomForms.length > 0) {
    printer.warn(`The following form${detachedCustomForms.length === 1 ? '' : 's'} will no longer be available because the connected data model no longer exists: ${detachedCustomForms.join(', ')}`);
  }
};
```